<?xml version="1.0"?>
<robot name="mini_turty">

  <link name="base_link">
    <visual>
      <geometry>
<!--        <cylinder length="0.1" radius="0.25"/>-->
        <cylinder length="0.05" radius="0.08"/>
      </geometry>
    </visual>
  </link>

  <link name="base_footprint">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
    </visual>
  </link>

  <joint name="base_to_footprint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>

  <link name="base_laser">
    <visual>
      <geometry>
<!--        <box size="0.1 0.05 0.2"/>-->
        <box size="0.045 0.045 0.056"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1.0"/>
      </material>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>

  <joint name="base_to_laser" type="fixed">
    <parent link="base_link"/>
    <child link="base_laser"/>
<!--    <origin rpy="0.0 0.0 0.0" xyz="-0.1 0.0 0.1"/>-->
<!--    <origin rpy="0.0 0.0 3.142" xyz="-0.1 0.0 0.1"/>-->
<!-- JJ - this is basically working with ydlidar and hector
    <origin rpy="0.0 0.0 0.0" xyz="-0.1 0.0 0.1"/> -->
<!-- JJ - changed to this to fix problem of Goal Setting in nav, set ydlidar 180 rotated on z-axis
    <origin rpy="0.0 0.0 3.1415" xyz="-0.1 0.0 0.1"/> -->
<!-- JJ - changed to this try to fix getting stuck on corners -->
    <origin rpy="0.0 0.0 3.1415" xyz="0.0 0.0 0.1"/>
  </joint>

<!--
  <link name="imu">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1.0"/>
      </material>
      <origin rpy="0.0 0.0 1.57075" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>

  <joint name="base_to_imu" type="fixed">
    <parent link="base_link"/>
    <child link="imu"/>
    <origin rpy="0.0 0.0 1.57075" xyz="0.0 0.05 0.25"/>
  </joint>
-->

</robot>
